openedx/core/djangoapps/edx_global_analytics/utils.py

```diff
@@ -0,0 +1,128 @@
+"""
+Helpers for the edX global analytics application.
+"""
+
+import datetime
+import calendar
+
+from django.core.cache import cache
+
+from django.db.models import Count
+from django.db.models import Q
+from student.models import UserProfile
+
+
+def calculate_active_students_amount(query_name, activity_period):
+    """
+    Calculates instance active students amount for particular period as like as previous calendar day,
+    week and month after cached.
+    """
+    period_start, period_end = activity_period
+
+    active_students_amount = UserProfile.objects.exclude(
+        Q(user__last_login=None) | Q(user__is_active=False)
+    ).filter(user__last_login__gt=period_start, user__last_login__lt=period_end).count()
```

Comment:
I researched Djangos logic for that, Ill implement __gte feature, it works fine.